zkvm/src/vm.rs

```diff
@@ -287,6 +287,31 @@ where
         Ok(())
     }
 
+    fn const_instr(&mut self) -> Result<(), VMError> {
+        let a = self.pop_item()?.to_data()?.to_scalar()?;
+        self.push_item(Expression::from(a));
+        Ok(())
+    }
+
+    fn var(&mut self) -> Result<(), VMError> {
+        let comm = self.pop_item()?.to_data()?.to_commitment()?;
+        let v = self.make_variable(comm);
+        self.push_item(v);
+        Ok(())
+    }
+
+    fn mintime(&mut self) -> Result<(), VMError> {
+        let a: Scalar = self.mintime.into();
+        self.push_item(Expression::from(a));
+        Ok(())
+    }
+
+    fn maxtime(&mut self) -> Result<(), VMError> {
+        let a: Scalar = self.maxtime.into();
+        self.push_item(Expression::from(a));
```

Comment:
seems like this PR does not define `Expression::from`. And should it be more explicit, like `Expression::constant(Scalar)`? Spec uses "constant expressions" as a guaranteed optimization, so it's good to have explicit API for these.

Reply:
Yep I agree